reflex/model.py

```diff
@@ -184,7 +192,7 @@ def run_autogenerate(rev, context):
             upgrade_ops = revision_context.generated_revisions[-1].upgrade_ops
             if upgrade_ops is not None:
                 changes_detected = bool(upgrade_ops.ops)
-        if changes_detected:
+        if changes_detected and write_migration_scripts:
```

Comment:
Below this line, can we just make `revision_context.generate_scripts()` generate all the scripts without having to iterate? Or are there places where we need the `_script` value?

Reply:
i ended up turning this into a `tuple()` collapse. the problem is that `generate_scripts` is an iterator, so we need to iterate through it or the scripts don't actually get written out.